### PR TITLE
Fixes #25569 - fixed windows os

### PR DESCRIPTION
**Pull request by ShimShtein**
This change is based on two templates (attached):
[windows_provisioning.txt](https://github.com/theforeman/foreman/files/2644041/windows_provisioning.txt)
[windows_pxe.txt](https://github.com/theforeman/foreman/files/2644042/windows_pxe.txt)

The provisioning template attempts to recode `root_pass` by using `encrypt_password` method from operating system class.
Right now the method is pass-through, and needs refactoring.

Changes made to the template:
`@host.param[x]` becomes `host_param(x)`
`@host.operatingsystem.bootfile(` now accepts `medium_provider` instead of `@host.architecture`

### CODE REVIEW

File: app/models/operatingsystems/windows.rb [hunk 1]
```diff
@@ -36,4 +40,9 @@ def boot_files_uri(medium_provider)
   def display_family
     "Windows"
   end
+
+  def encrypt_password(root_pass)
```
Comment (tbrisker) [inline app/models/operatingsystems/windows.rb]: Since this is an implementation detail in the specific template, i would rather exclude it from here and only fix the "official" api to work (and cp to 1.20-stable)

**Review by tbrisker — CHANGES_REQUESTED** [review]

**Comment by UXabre** [comment]
Not sure if this is the right place to discuss this, but for other operatingsystems, I've also come across "tiny" (hash) differences that are not compatible between different OS implementations. In this case it's base-64 for windows, in my case it's SHA-256 for Huawei switches. 

I've been contemplating if it would not be wise to allow an OperatingSystem to provide a list of supported encryption methods (and fallback to the current, default, list of available hashing methods) and a way to override their specific implementation if needed as well.
This way, one could have Base-64 specific implementation available on windows but without having to resort to contaminating the total list of encryption methods. The same applies for Huawei and probably others as well. This would keep things clean and lower the amount of total choices which prevents errors/mistakes/slips in the end.

**Comment by ShimShtein** [comment]
@UXabre, created an RFC on [discourse](https://community.theforeman.org/t/rfc-move-admin-password-encoding-to-os-object/12301).

TL;DR: :+1: from me. It's a good idea moving this responsibility to `OperatingSystem` object.

**Comment by iNecas** [comment]
What's status of this PR?

**Review by tbrisker — APPROVED** [review]
I'm fine with merging this for now to unblock windows provisioning and refactoring later to match the algorithms to the os

**Comment by UXabre** [comment]
Makes sense for me as well, currently the need for this refactoring is non mandatory so we can do it at a latter stage as well IMHO :-)

**Comment by tbrisker** [comment]
1.21 - c1c4d5dd5a06cb533cce8e3b25e7f9625f8eccfb
1.20 - fd9352ead
